src/k8s/api/v1/types.go

```diff
@@ -116,15 +116,12 @@ func (c ClusterStatus) String() string {
 
 	// Control Plane Nodes
 	result.WriteString(fmt.Sprintf("%-*s ", maxLen, "control plane nodes:"))
-	addrMap := c.getCPNodeAddrToRoleMap()
-	nodes := make([]string, len(addrMap))
-	i := 0
-	for addr, role := range addrMap {
-		nodes[i] = fmt.Sprintf("%s (%s)", addr, role)
-		i++
-	}
-	if len(nodes) > 0 {
-		result.WriteString(strings.Join(nodes, ", "))
+	if len(c.Members) > 0 {
+		for _, m := range c.Members {
+			if m.ClusterRole == ClusterRoleControlPlane {
+				result.WriteString(fmt.Sprintf("%s (%s)  ", m.Address, m.DatastoreRole))
+			}
+		}
```

Comment:
put this into an array and `strings.Join(parts, ", ")` at the end, let's not add whitespace at end of lines